producer_test.go

```diff
@@ -410,4 +438,126 @@ func testProducer(t *testing.T, makeProducer func(ctx context.Context, t *testin
 
 		startstoptest.Stress(ctx, t, producer)
 	})
+
+	t.Run("QueuePausedBeforeStart", func(t *testing.T) {
+		t.Parallel()
+
+		producer, bundle := setup(t)
+		AddWorker(bundle.workers, &noOpWorker{})
+
+		// TODO: maybe add a separate QueueInsertFull to driver to clean this up?
```

Comment:
IMO: while there's some argument that jobs should have a fast path because we want them highly performant for benchmarks and the like, one extra insert parameter on queues is going to have ~zero practical performance impact. Maybe just add any desired ones to the normal insert path as optional parameters.

Reply:
Yep great call, fixed and cleaned up a few places that were separately re-pausing (except those which were explicitly testing the act of pausing).